### PR TITLE
[close #265] support unsafeDestoryRange for KVClient

### LINKED ISSUE #265: Propose: support UnsafeDestroyRange API
We need this API to quickly clean up data, hoping to support. We can contribute this feature, thanks!

### PR DESCRIPTION
**Pull request by gotoxu**
Signed-off-by: Xu Qiaolun <Jamesxql@Gmail.com>

fix #265

### CODE REVIEW

**Comment by marsishandsome** [comment]
@gotoxu please rebase master

**Comment by gotoxu** [comment]
@marsishandsome done

**Review by marsishandsome — COMMENTED** [review]
please add some tests

**Comment by marsishandsome** [comment]
@gotoxu please add DCO 
image

File: src/test/java/org/tikv/txn/KVClientTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    if (session != null) {
+      session.close();
+    }
+  }
+
+  @Test
```
Comment (marsishandsome) [inline src/test/java/org/tikv/txn/KVClientTest.java:33]:
```suggestion
  @Ignore
```

Comment (marsishandsome) [reply src/test/java/org/tikv/txn/KVClientTest.java:33]: Current CI only support run RawKV test, cause with TTL enabled TxnKV API is not supported.

```
[2021-09-22T02:51:20.675Z] [2021/09/22 10:41:19.973 +08:00] [WARN] [session.go:1579] ["run statement failed"] [schemaVersion=0] [error="tikv aborts txn: Error(Other(\"[src/storage/mod.rs:952]: can't sched txn cmd(prewrite) with TTL enabled\"))"] [session="{\n  \"currDBName\": \"\",\n  \"id\": 0,\n  \"status\": 2,\n  \"strictMode\": true,\n  \"user\": null\n}"]

[2021-09-22T02:51:20.675Z] goroutine 1 [running]:

[2021-09-22T02:51:20.675Z] runtime/debug.Stack(0x0, 0xc000f9f758, 0x0)

[2021-09-22T02:51:20.675Z] 	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f

[2021-09-22T02:51:20.675Z] runtime/debug.PrintStack()

[2021-09-22T02:51:20.675Z] 	/usr/local/go/src/runtime/debug/stack.go:16 +0x25

[2021-09-22T02:51:20.675Z] github.com/pingcap/tidb/session.mustExecute(0x4243ed8, 0xc0000ff400, 0x3cdfa6a, 0x22, 0x0, 0x0, 0x0)

[2021-09-22T02:51:20.675Z] 	/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/session/bootstrap.go:1742 +0xd9

[2021-09-22T02:51:20.675Z] github.com/pingcap/tidb/session.doDDLWorks(0x4243ed8, 0xc0000ff400)

[2021-09-22T02:51:20.675Z] 	/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/session/bootstrap.go:1604 +0x5f

[2021-09-22T02:51:20.675Z] github.com/pingcap/tidb/session.bootstrap(0x4243ed8, 0xc0000ff400)

[2021-09-22T02:51:20.675Z] 	/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/session/bootstrap.go:373 +0x353

[2021-09-22T02:51:20.675Z] github.com/pingcap/tidb/session.runInBootstrapSession(0x4206650, 0xc00041a540, 0x3da3d28)

[2021-09-22T02:51:20.675Z] 	/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/session/session.go:2579 +0x297

[2021-09-22T02:51:20.675Z] github.com/pingcap/tidb/session.BootstrapSession(0x4206650, 0xc00041a540, 0x0, 0x0, 0x0)

[2021-09-22T02:51:20.675Z] 	/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/session/session.go:2432 +0xd1b

[2021-09-22T02:51:20.675Z] main.createStoreAndDomain(0x5fb89c0, 0x3d0289c, 0x2c)

[2021-09-22T02:51:20.675Z] 	/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/tidb-server/main.go:290 +0x189

[2021-09-22T02:51:20.675Z] main.main()

[2021-09-22T02:51:20.675Z] 	/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/tidb-server/main.go:191 +0x26a

[2021-09-22T02:51:20.675Z] [2021/09/22 10:41:19.973 +08:00] [FATAL] [bootstrap.go:1743] ["mustExecute error"] [error="tikv aborts txn: Error(Other(\"[src/storage/mod.rs:952]: can't sched txn cmd(prewrite) with TTL enabled\"))"] 
```

**Review by marsishandsome — APPROVED** [review]
LGTM

**Comment by zz-jason** [comment]
@gotoxu please fix license check failure:
```
ERROR the following files don't have a valid license header: 
src/test/java/org/tikv/txn/KVClientTest.java 
```

**Review by zz-jason — APPROVED** [review]
LGTM

File: src/main/java/org/tikv/txn/KVClient.java [hunk 2]
```diff
@@ -195,6 +205,64 @@ public synchronized void ingest(List<Pair<ByteString, ByteString>> list) throws
     }
   }
 
+  public void unsafeDestoryRange(ByteString startKey, ByteString endKey) throws GrpcException {
```
Comment (zz-jason) [inline src/main/java/org/tikv/txn/KVClient.java:208]: please add a unit test for this function to improve the code coverage.

File: src/main/java/org/tikv/txn/KVClient.java [hunk 3]
```diff
@@ -195,6 +205,64 @@ public synchronized void ingest(List<Pair<ByteString, ByteString>> list) throws
...
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new TiKVException("Current thread interrupted.", e);
+    } catch (ExecutionException e) {
+      throw new TiKVException("Execution exception met.", e);
+    }
+  }
+
+  private Object doUnsafeDestoryRange(TiStore store, ByteString startKey, ByteString endKey) {
```
Comment (zz-jason) [inline src/main/java/org/tikv/txn/KVClient.java:230]: ditto

**Comment by zz-jason** [comment]
I'm going to close this PR since it hasn't been updated for a long time, feel free to reopen it in the future once needed.

**Pull request closed without merging**